BSP/STM32F413/Src/BSP_Contactor.c

```diff
@@ -1,64 +1,105 @@
 /* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
 
 #include "BSP_Contactor.h"
-#include "stm32f4xx.h"
 
 /**
- * @brief   Initializes the GPIO pins that interfaces with the Contactor.
- *          Two GPIO pins are initialized. One as an output and one as an input.
- *          The output pin controls the state and the input pin views what state the contactor is through the
- *          Aux pins.
- * @param   None
- * @return  None
+ * @brief Initializes all Contactor pins used by the BPS
+ * @param none
+ * @return none
  */
 void BSP_Contactor_Init(void) {
-    GPIO_InitTypeDef GPIO_InitStruct;
-	
-	// Initialize clock
-	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE);
-	
-	// Initialize PB0 as output
-	GPIO_InitStruct.GPIO_Pin = GPIO_Pin_0;			// Pin 0 is output
-	GPIO_InitStruct.GPIO_Mode = GPIO_Mode_OUT;
-	GPIO_InitStruct.GPIO_Speed = GPIO_Speed_50MHz;
-	GPIO_InitStruct.GPIO_OType = GPIO_OType_PP;
-	GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
-	GPIO_Init(GPIOB, &GPIO_InitStruct);
-	GPIO_InitStruct.GPIO_Pin = GPIO_Pin_1;			// Pin 1 is input
-	GPIO_InitStruct.GPIO_Mode = GPIO_Mode_IN;
-	GPIO_Init(GPIOB, &GPIO_InitStruct);
+	//since pwm was setup in BSP_PWM_INIT, all we need to do is setup an input pin
+    GPIO_InitTypeDef GPIO_C1Init;
```

Comment:
Call PWM Init here anyways, just in case it wasn't called before (I don't see it called anywhere btw).

Reply:
oop I meant to call it somewhere else but here works too